m4/ax_boost_base.m4

```diff
@@ -126,8 +126,14 @@ AC_DEFUN([_AX_BOOST_BASE_RUNDETECT],[
       [multiarch_libsubdir="lib/${host_cpu}-${host_os}"]
     )
 
+    dnl some arches may advertise a cpu type that doesn't line up with their
+    dnl prefix's cpu type. For example, uname may report armv7l while libs are
+    dnl installed to /usr/lib/arm-linux-gnueabihf. Try getting the compiler's
+    dnl value for an extra chance of finding the correct path.
+    libsubdirs="lib/`$CXX -dumpmachine 2>/dev/null` $libsubdirs"
```

Comment:
I doubt that this portable. GNU GCC will respond properly to that command line option, but what about other compilers?

Reply:
FWIW clang 10 does as well, at least
```
$ /opt/clang10/bin/clang -dumpmachine
x86_64-unknown-linux-gnu
```
but I agree that this needs error handling, the output of `-dumpmachine` should only be added when the command succeeds

Reply:
ndk (clang based android native dev kit) doesn't seem to work on 0.19.0.1 (but does on 0.18.1) - testing a bit

Reply:
This is not portable on ARMv7 with clang:
```
$ clang++ --version
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
Target: armv8l-unknown-linux-gnueabihf
Thread model: posix
InstalledDir: /usr/bin
$ clang++ -dumpmachine
armv8l-unknown-linux-gnueabihf
```

For comparison:
```
$ g++ --version | head -1
g++ (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0
$ g++ -dumpmachine
arm-linux-gnueabihf
```

```